### PR TITLE
723 consolidate processing

### LINKED ISSUE #543: Documents should be viewable before processing finishes
With PDF.js, we'll be able to view a document as soon as it's uploaded. This raises some questions:

- I'm not sure if that has privacy implications (since the original file is in private storage until it's done processing and marked public). 
- I also don't know what other parts of the UI will depend on full processing.
- If there's an error in processing, we might need to nudge people to reprocess.

### PR DESCRIPTION
**Pull request by eyeseast**
- [x] pending document status
- [x] reprocess a document
- [x] running add-on status
- [x] dismiss add-on runs
- [x] cancel a running add-on
- [x] final design

Closes #543, #646, #652, #723

### CODE REVIEW

**Comment by eyeseast** [comment]
Mechanics are done on this. Just needs final design. I have two separate components now, both on the search and document routes for testing, but they could be combined and placed somewhere else as needed.

**Comment by allanlasser** [comment]
- Dispatching an add-on run doesn't seem to populate the store with processing state.
- Reprocessing a document leads to uncaught 404 error for document.txt.json, which blocks further UI updates.

File: src/lib/components/processing/AddOns.svelte [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    if ($running.length === 0) {
+      await load();
+    }
+  });
+
+  afterUpdate(() => {
+    if ($running.length > 0) {
+      timeout = setTimeout(load, POLL_INTERVAL);
+    }
+  });
```
Comment (allanlasser) [inline src/lib/components/processing/AddOns.svelte:41]: This won't check for dispatched addons if there are not any currently running. How do we make sure that in a clean session, dispatching an add-on populates the `$running` store?

**Review by allanlasser — APPROVED** [review]
OK, everything is working well!

I think there are some UX enhancements we could make to optimistically add new processing items and dismiss add-ons so that it feels snappier, but those would build on this foundation. Good stuff!

**Pull request merged**